hazelcast/src/main/java/com/hazelcast/sql/impl/type/converter/BooleanConverter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.type.converter;
+
+import com.hazelcast.sql.impl.type.QueryDataTypeFamily;
+
+/**
+ * Converter for {@link java.lang.Boolean} type.
+ */
+public final class BooleanConverter extends Converter {
+
+    public static final BooleanConverter INSTANCE = new BooleanConverter();
+
+    protected static final String TRUE = "true";
+    protected static final String FALSE = "false";
+
+    private BooleanConverter() {
+        super(ID_BOOLEAN, QueryDataTypeFamily.BIT);
+    }
+
+    @Override
+    public Class<?> getValueClass() {
+        return Boolean.class;
+    }
+
+    @Override
+    public boolean asBit(Object val) {
+        return ((Boolean) val);
+    }
+
+    @Override
+    public String asVarchar(Object val) {
+        boolean val0 = (Boolean) val;
+
+        return val0 ? TRUE : FALSE;
```

Comment:
Looks like it might be `null` as well.

Reply:
There are no null-checks throughout converters code, we assume that the value passed to the converter should never be `null`. null-checks may degrade performance because most expressions have special treatment for null-s, so null-check is already performed before the converter is called. Two alternative solutions:
1. Have null-checks in both expressions and converters - bad, since these are very performance-critical places
1. Have null-checks in converters only - doesn't work, since expression must process null-s separately anyway